app/controllers/katello/concerns/organizations_controller_extensions.rb

```diff
@@ -29,6 +29,7 @@ def create_with_dynflow
             @taxonomy = Organization.new(resource_params)
             sync_task(::Actions::Katello::Organization::Create, @taxonomy)
             @taxonomy.reload
+            session[:organization_id] = @taxonomy.id
```

Comment:
>  that would mean creating a new action in the controller and additional redirect

No, it would mean a redirect to `/organizations/2-work/select` which is served by [taxonomies_controller](https://github.com/theforeman/foreman/blob/develop/app/controllers/concerns/foreman/controller/taxonomies_controller.rb#L98), that is how it's done in the UI.
By the way, that action contains actually all the code necessary to switch to new org.